powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java

```diff
@@ -258,23 +257,33 @@ public OperationDetailResponse createOperation(OperationCreateRequest request) t
         }
     }
 
+    private String fetchUserId(final OperationCreateRequest request) throws GenericServiceException {
+        if (request.getUserId() != null) {
+            return request.getUserId();
+        } else if (request.getActivationId() != null) {
+            final String activationId = request.getActivationId();
+            logger.debug("Filling missing user ID from the activation ID: {}", activationId);
+            return activationRepository.findById(activationId)
+                    .map(ActivationRecordEntity::getUserId)
```

Comment:
This should not be null, like ever - if no activation exists, it will raise an exception later in the code.

Reply:
Ok, let's throw an exception.